File: python/baekjoon/1912.py
# Created by Jello on 2017. 7. 1.

T = int(input())
seq = list(map(int, str(input()).split(' ')))
result = sum = seq[0]

for i in range(1, T):
    sum = max(sum+seq[i], seq[i])
    result = max(result, sum)

print(result)

File: c++/baekjoon/1912.cpp
#include <iostream>
#include <vector>
using namespace std;

int main(){
    int T = 0;
    cin >> T;
    vector<int> arr(T);
    int sum, result;
    sum = result = 0;

    for(int i=0;i<T;i++){
        cin >> arr[i];
        if(i == 0) sum = result = arr[i];
        else{
            sum = sum+arr[i] > arr[i] ? sum+arr[i] : arr[i];
            result = result > sum ? result : sum;
        }
    }

    cout << result;
}